glass-easel/src/shadow_root.ts

```diff
@@ -47,23 +47,23 @@ export class ShadowRoot extends VirtualNode {
   private _$host: GeneralComponent
   /** @internal */
   _$backendShadowRoot: backend.ShadowRootContext | null
+  private _$slotMode: SlotMode
   private _$idMap: { [id: string]: Element } | null
-  private _$slotCacheDirty: boolean
-  private _$appliedSlotCacheDirty: boolean
   private _$singleSlot?: Element | null
-  private _$appliedSingleSlot?: Element | null
-  private _$slots?: { [name: string]: Element }
-  private _$appliedSlots?: { [name: string]: Element }
-  private _$dynamicSlotSet?: Set<Element>
-  private _$appliedDynamicSlots?: Map<Element, AppliedSlotMeta>
+  private _$slots?: Record<string, Element>
+  private _$slotsList?: Record<string, DoubleLinkedList<Element>>
+  private _$dynamicSlotsInserted?: boolean
+  private _$dynamicSlots?: Map<Element, AppliedSlotMeta>
   private _$requiredSlotValueNames?: string[]
   private _$propertyPassingDeepCopy?: DeepCopyStrategy
   private _$insertDynamicSlotHandler?: (
-    slot: Element,
-    name: string,
-    slotValues: { [name: string]: unknown },
+    slots: {
```

Comment:
Why changed to an array?

Reply:
> Why changed to an array?

Changed it to array is for batch these slot operations, as batching operations can reduce backend operations and avoid complexity from growing caused by operation spreading.

Reply:
But backend operations are always batched?